packages/gestalt/src/Toast.js

```diff
@@ -107,16 +124,28 @@ export default function Toast({
               </Mask>
             </Flex.Item>
           ) : null}
-
           <Flex.Item flex="grow" maxWidth={hasImage && hasButton ? TEXT_MAX_WIDTH_PX : undefined}>
-            <Text align={!thumbnail && !button ? 'center' : 'start'} color={textColor}>
+            <Text align={!thumbnail && !primaryAction ? 'center' : 'start'} color={textColor}>
               {textElement}
             </Text>
           </Flex.Item>
-
-          {button ? (
+          {primaryAction || _dangerouslySetPrimaryAction ? (
             // Allow button text to wrap on mobile
-            <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>{button}</Flex.Item>
+            <Flex.Item flex={isMobileWidth ? 'shrink' : 'none'}>
+              {isValidElement(_dangerouslySetPrimaryAction) ? _dangerouslySetPrimaryAction : null}
+              {!_dangerouslySetPrimaryAction &&
+              primaryAction?.accessibilityLabel &&
```

Comment:
I like that we're making these checks, but I think we should move them to ToastPrimaryAction, where we can throw an actual error. Flow will already be yelling at the developer about missing fields; throwing an error would accomplish the same thing you're doing here, but with more direct feedback about what's missing and how to fix it.

Reply:
I can set an error to double down on the user feedback, though Flow is already covering it